pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java

```diff
@@ -106,63 +104,43 @@ public void initialize(ServiceConfiguration conf, PulsarResources pulsarResource
-                        case Prefix:
-                            if (!subscription.startsWith(role)) {
-                                PulsarServerException ex = new PulsarServerException(String.format(
-                                        "Failed to create consumer - The subscription name needs to be prefixed by the "
-                                                + "authentication role, like %s-xxxx for topic: %s", role, topicName));
-                                permissionFuture.completeExceptionally(ex);
-                                return;
+                            // validate if subscription-auth mode is configured
+                            if (policies.get().subscription_auth_mode != null) {
+                                switch (policies.get().subscription_auth_mode) {
+                                    case Prefix:
+                                        if (!subscription.startsWith(role)) {
+                                            PulsarServerException ex = new PulsarServerException(String.format(
+                                                 "Failed to create consumer - The subscription name needs to be"
+                                                 + " prefixed by the authentication role, like %s-xxxx for topic: %s",
+                                                 role, topicName));
+                                            return FutureUtil.failedFuture(ex);
+                                        }
+                                        break;
+                                    default:
+                                        break;
+                                }
                             }
-                            break;
-                        default:
-                            break;
                         }
                     }
-                }
-                // check namespace and topic level consume-permissions
-                checkAuthorization(topicName, role, AuthAction.consume).thenAccept(isAuthorized -> {
-                    permissionFuture.complete(isAuthorized);
-                }).exceptionally(ex -> {
-                    log.warn("Client with Role - {} failed to get permissions for topic - {}. {}", role, topicName,
-                            ex.getMessage());
-                    permissionFuture.completeExceptionally(ex);
-                    return null;
+                    return checkAuthorization(topicName, role, AuthAction.consume);
                 });
-            }).exceptionally(ex -> {
-                log.warn("Client with Role - {} failed to get permissions for topic - {}. {}", role, topicName,
-                        ex.getMessage());
-                permissionFuture.completeExceptionally(ex);
-                return null;
-            });
-        } catch (Exception e) {
-            log.warn("Client  with Role - {} failed to get permissions for topic - {}. {}", role, topicName,
```

Comment:
After this change, we will remove this log? We'd better keep the behavior will not be changed.

Reply:
ok, done.